### PR TITLE
CoPR: Autobuild rpm on rhcontainerbot/podman-next

### PR DESCRIPTION
**Pull request by lsm5**
The new file `skopeo.spec.rpkg` along with a webhook will automatically
build rpms on every PR merge on the main branch.

Run `rpkg local` or `make rpm` to generate the rpm.

Known issue: Doesn't yet build for EL8 environments.

Signed-off-by: Lokesh Mandvekar <lsm5@fedoraproject.org>

@rhatdan @vrothberg @mtrmac PTAL

I have added the webhook already, so merging this should trigger an rpm build on the copr.

### CODE REVIEW

File: skopeo.spec.rpkg [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# For automatic rebuilds in COPR
```
Comment (mtrmac) [inline skopeo.spec.rpkg:1]: A totally non-blocking it: Would it be (_without extra hassle_) possible to move this somewhere to `contrib`? We don’t really want users to build their own RPMs, and that’s where the container image descriptions are (admittedly `contrib` is not a great name for “our own build infrastructure that other users shouldn’t need to touch”).

OTOH I see `podman.spec.rpkg` in the same place, and that consistency is certainly valuable. So entirely up to you.

**Review by mtrmac — COMMENTED** [review]
Thanks.

Is this related to #1661 ? I’m rather confused about how that is supposed to work — the original issue is that building cross-architecture containers using qemu-user-static takes too long, and the improvement is to first build RPMs and then build containers using those RPMs, in two separate stages and systems?

How is that faster? Is there some significant infrastructure difference, like the RPM builders being native, or the hardware being faster/cheaper?

Or is this completely unrelated to #1681?

**Comment by lsm5** [comment]
> Thanks.
> 
> Is this related to #1661 ? I’m rather confused about how that is supposed to work — the original issue is that building cross-architecture containers using qemu-user-static takes too long, and the improvement is to first build RPMs and then build containers using those RPMs, in two separate stages and systems?
> 
> How is that faster? Is there some significant infrastructure difference, like the RPM builders being native, or the hardware being faster/cheaper?
> 
> Or is this completely unrelated to #1681?

Completely unrelated, unless @cevich wants it to be. This is mainly a way to let users to fetch RPMs built using the latest merged upstream code. With the current setup, this will only build RPMs `post-merge`. There might be ways to trigger it pre-merge for testing purposes but I don't yet have a good way to do it without polluting the CoPR with a ridiculous amount of builds.

File: skopeo.spec.rpkg (hunk 1 above)
Comment (lsm5) [reply skopeo.spec.rpkg:1]: Let me check if that's doable. IIRC, this file is expected in the main dir, but let me confirm..

**Comment by mtrmac** [comment]
> This is mainly a way to let users to fetch RPMs built using the latest merged upstream code. 

Thanks. I’ll entirely defer to your expertise, then — feel free to merge as appropriate.

**Comment by lsm5** [comment]
Thanks @mtrmac. Merging ..

**Pull request merged**